lib/middleware/vhost.js

```diff
@@ -15,9 +17,9 @@ module.exports = () =>
       if (bucket) {
         ctx.path = `/${bucket}${ctx.path}`;
       }
-    } else if (ctx.hostname !== "localhost" && ctx.hostname !== "127.0.0.1") {
+    } else if (!net.isIP(ctx.hostname) && ctx.hostname !== "localhost") {
       // otherwise attempt to distinguish virtual host-style requests
-      ctx.path = `/${bucket}${ctx.path}`;
+      ctx.path = `/${ctx.hostname}${ctx.path}`;
```

Comment:
hmmm I will retest tomorrow but in my case i was going in that if and it was working fine with `ctx.path = bucket/path`